presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.List;
+
+public interface RowEncoder
+{
+    ProducerRecord<Object, Object> encodeRow(List<EncoderFieldValueProvider> rowData, String topicName);
+
+    String getSerializer();
```

Comment:
Instead of passing class by name, we could pass actual objects, if we invoked `org.apache.kafka.clients.producer.KafkaProducer#KafkaProducer(java.util.Properties, org.apache.kafka.common.serialization.Serializer<K>, org.apache.kafka.common.serialization.Serializer<V>)` -- `KafkaProducer` constructor overload.

The gain is not in terms of saving reflective call for performance reasons, but to increase flow visibility within IDE.